### PR TITLE
move imports to body in notebooks

### PR DESCRIPTION
**Pull request by joelberkeley-secondmind**
When I learnt how to use numpy, pandas etc., the tutorials only imported libraries when it used them, rather than at the start of the notebook. This made it clear what is needed when and why. In this PR I've done that to the notebooks.

### CODE REVIEW

File: docs/notebooks/failure_ego.pct.py [hunk 1]
```diff
@@ -17,27 +17,9 @@
...
-from trieste.models.model_interfaces import VariationalGaussianProcess
-from trieste.models import ModelSpec
-from gpflow.models import VGP
-from gpflow.optimizers import NaturalGradient
 
-from util.plotting_plotly import plot_function_plotly, plot_gp_plotly, add_bo_points_plotly
-from util.plotting import plot_gp_2d, plot_function_2d, plot_bo_points
-
-# %%
-gpflow.config.set_default_jitter(1e-5)
```
Comment (joelberkeley) [inline docs/notebooks/failure_ego.pct.py:24]: this particular line change is outside the scope of this PR. I couldn't resist, cos it makes the intro to all the notebooks the same. I checked and the notebook results are the same

**Review by vpicheny — APPROVED** [review]
LGTM
